ethereumj-core/src/main/java/org/ethereum/sync/FastSyncManager.java

```diff
@@ -154,6 +154,10 @@ public synchronized void reqSent(Long requestId) {
             requestSent.put(requestId, timestamp);
         }
 
+        public synchronized Set<Long> requestIdsSnapshot() {
```

Comment:
To avoid sync problems with `requestSent` and eliminate additional locks you need to synchronize here on a `FastSyncManager` instance:
```
synchronize(FastSyncManager.this) {
}
```